### ISSUE TITLE
Fix doctests on GHC 8.10 and GHC 9.0

### ISSUE THREAD

**Issue by Shimuuar** [haskell/vector#370]
At least on CI they don't work at all and fail with something like that:

```
Data/Vector/Storable.hs:558: failure in expression `import qualified Data.Vector.Storable as VS'
expected: 
 but got: 
          ^
          <interactive>:1:1: error:
              attempting to use module ‘main:Data.Vector.Storable’ (Data/Vector/Storable.hs) which is not loaded
```

Cursory search doctest's issue tracker does reveal anything interesting

**Comment by Shimuuar** [comment]
I tried to run doctests on 8.10 locally. There're weird error when something is not in scope when it was imported  line above. On top of that they become very slow. First 141 test pass very quickly after that they slow to a crawl

Situation with doctests certainly doesn't look good.

**Comment by Bodigrim** [comment]
I lost my hope in doctests, they are so difficult to setup properly on recent GHCs :(

**Comment by Shimuuar** [comment]
They probably couldn't work properly without explicit support from GHC

**Comment by lehins** [comment]
I just disabled doctests in `random` for newer ghc: https://github.com/haskell/random/blob/b89d4117797e646ab55a8fece5b468a2d5b10682/test/doctests.hs#L4-L10

So, I agree situation with doctests is not great. However, I think it is important to run doctests at least on one ghc version. The point of doctests is not to test the underlying representation, that's what unit, property ... tests are for, but to make sure that examples in documentation are still correct.

I suggest we do the same as I did in `random`. Disable doctests on newer ghc versions, until situation improves. Maybe that could be one of those things that Haskell Foundation could help find people interested in figuring out a solution.
